### PR TITLE
feat(tianmu): support alter table xxx modify xxx type NULL/NOT NULL(#860)

### LINKED ISSUE #860: feature(tianmu): support `alter table xxx modify xxx type(length) default null`
**Is your feature request related to a problem? Please describe.**

Enable the tianmu engine to support syntax: ```alter table xxx modify xxx type(length) default null```
**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by DandreChen**
## Summary about this PR

Issue Number: close #860 

[summary]
support field properties from NULL to NOT NULL
support field properties from NOT NULL to NULL

## Tests Check List

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [x] New Feature
- [ ] Bug Fix
- [ ] Improvement
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

**Comment by hustjieke** [comment]
> This pull request's title should follow requirements next. @DandreChen please check it 👇.
> 
> Valid format:
> 
> ```
> fix(vc): fix sth..... (#3306)
>   ^         ^---------^  ^----^
>   |         |            |
>   |         +            +-> you issue id.
>   |         |
>   |         +-> Summary in present tense.
>   |
>   +-------> Type: feat, fix, docs, workflow, style, refactor, test, website, chore
> ```
> 
> Valid types:
> 
> * `feat`: new feature for stonedb
> * `fix`: bug fix for stonedb
> * `docs`: changes to the documentation
> * `workflow`: ci/cd in .github
> * `refactor`: refactoring production code, eg. renaming a variable
> * `style`: formatting, missing semi colons, etc; no production code change
> * `test`: adding missing tests, refactoring tests; no production code change
> * `website`
> * `chore`: updating grunt tasks etc; no production code change

@DandreChen pls fix the title.

File: mysql-test/suite/tianmu/t/alter_table_null.test [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  `c_longblob` longblob COMMENT 'longblob'
+) ENGINE=TIANMU DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+#
+#ALTER TABLE: NOT NULL to NULL 
+#
+
+show create table alter_table_null_test;
+
+alter table alter_table_null_test modify c_smallint smallint NOT NULL;
```
Comment (lujiashun) [inline mysql-test/suite/tianmu/t/alter_table_null.test:35]: this case is  'not null' to "not null", lack "not null" to "null" case;

**Review by RingsC — APPROVED** [review]

**Review by hustjieke — APPROVED** [review]
LGTM

File: mysql-test/suite/tianmu/t/alter_table_null.test (hunk 1 above)
Comment (DandreChen) [reply mysql-test/suite/tianmu/t/alter_table_null.test:35]: fixed it

**Pull request merged**